Auto-generate yet another handful of WebCore/platform coders

beidson opened the pull request:
#### d2bfb96ad96328fcbdb30379b6ed05015ec4aa51
<pre>
Auto-generate yet another handful of WebCore/platform coders
<a href="https://bugs.webkit.org/show_bug.cgi?id=262090">https://bugs.webkit.org/show_bug.cgi?id=262090</a>
rdar://116027783

Reviewed by Alex Christensen.

* Source/WebCore/platform/ThreadSafeDataBuffer.h:
(WebCore::ThreadSafeDataBufferImpl::create):
(WebCore::ThreadSafeDataBuffer::create):
(WebCore::ThreadSafeDataBuffer::ThreadSafeDataBuffer):
(WebCore::ThreadSafeDataBuffer::encode const): Deleted.
(WebCore::ThreadSafeDataBuffer::decode): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: <a href="https://commits.webkit.org/268451@main">https://commits.webkit.org/268451@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/55cf906282e0eb9c4ed936dc66dbabaabe9dc5a2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/19730 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/20153 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/20770 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/21625 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/18437 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/19966 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/23413 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/20296 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/20007 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/19950 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/19953 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/17160 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/22479 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/17126 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/17954 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/24234 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18201 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/18128 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22226 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/18719 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/15863 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/17881 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4721 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/22232 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/18544 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebCore/platform/ThreadSafeDataBuffer.h:
```diff
@@ -56,7 +66,14 @@ friend class ThreadSafeDataBuffer;
+private:
+    friend struct IPC::ArgumentCoder<ThreadSafeDataBuffer, void>;
 public:
+    static ThreadSafeDataBuffer create(RefPtr<ThreadSafeDataBufferImpl>&& impl)
```
This shouldn't be public if it's only used by the ArgumentCoder, which we have befriended.

achristensen07 commented on Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
```diff
@@ -2560,6 +2560,11 @@ class WebCore::TransformOperations {
 }
 
+[Nested, AdditionalEncoder=StreamConnectionEncoder] struct WebCore::SourceBrush::Brush::LogicalGradient {
+    std::variant<Ref<WebCore::Gradient>, WebCore::RenderingResourceIdentifier> gradient;
```
I'm pretty sure we used to be sending a std::optional<WebCore::RenderingResourceIdentifier>.  Is there a conscious reason why that changed?

beidson replied:
The type is:
std::variant<Ref<Gradient>, RenderingResourceIdentifier>

I don't see where there's an optional<RenderingResourceIdentifier> either before or after.

That said, looking more closely *did* reveal a weird asymmetry in the hand written encoder/decoder. I've asked the original author about that on Slack.

achristensen07 commented on Source/WebCore/platform/ThreadSafeDataBuffer.h:
```diff
@@ -56,6 +64,8 @@ friend class ThreadSafeDataBuffer;
 };
 
 class ThreadSafeDataBuffer {
+private:
```
nit: not needed, since class has private by default.  Also, friend can go below with the other private stuff.

beidson committed:
Auto-generate yet another handful of WebCore/platform coders
https://bugs.webkit.org/show_bug.cgi?id=262090
rdar://116027783

Reviewed by Alex Christensen.

* Source/WebCore/platform/ThreadSafeDataBuffer.h:
(WebCore::ThreadSafeDataBufferImpl::create):
(WebCore::ThreadSafeDataBuffer::create):
(WebCore::ThreadSafeDataBuffer::ThreadSafeDataBuffer):
(WebCore::ThreadSafeDataBuffer::encode const): Deleted.
(WebCore::ThreadSafeDataBuffer::decode): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: https://commits.webkit.org/268451@main
